### DISCUSSION TITLE
Fast Refresh is not working in version 10.0.7 (Windows 10)

### DISCUSSION THREAD

**Discussion by DannyFeliz** [vercel/next.js#22214] [Help]
**What version of Next.js are you using?**

10.0.7

**What version of Node.js are you using?**

12.18.3

**What browser are you using?**

Chrome 88.0.4324.150 (Official Build) (64-bit)

**What operating system are you using?**

Windows 10 Pro Build 19042.804

**How are you deploying your application?**

next dev

**Describe the Bug**

I created a new project using `create-next-app` and Fast Refresh does not work. To see the changes, I have to reload the page manually. It doesn't work either in the WSL2.

It works fine if I use version 10.0.5.

**Expected Behavior**

Once I save my changes, the page should reload to reflect my changes.

**To Reproduce**

You can use this project to reproduce the issue: [next-fast-refresh-repo](https://github.com/DannyFeliz/next-fast-refresh-repo)

![fast](https://user-images.githubusercontent.com/5460365/108032878-839c3080-7009-11eb-8686-ccded067fcf2.gif)

**Comment by sebin-simon** [comment]
I have the same issue

**Comment by michaelwindell26** [comment]
I can confirm [the repo](https://github.com/DannyFeliz/next-fast-refresh-repo) does not fast refresh at all as shown in @DannyFeliz video.

I am on Ubuntu WSL2 with node 14.

Have you created an issue for this @DannyFeliz ?

**Reply by DannyFeliz** [reply]
@michaelwindell26 This used to be an issue; they converted it to a discussion for some reason

**Reply by wtatum** [reply]
I see the same in both 10.0.6 and 10.0.7, Windows 10, node 15.3

Totally clean `yarn create next-app` will not hot reload on 10.0.7 or 10.0.6 but works fine on 10.0.5

**Reply by DannyFeliz** [reply]
@wtatum please vote this discussion 
![image](https://user-images.githubusercontent.com/5460365/110013768-6d27f180-7cf8-11eb-8b1e-7497ae132e4b.png)

**Reply by charristti** [reply]
I am seeing the same problem. Fast Refresh broken starting at 10.0.6 -> 10.0.9. Testing using the tutorial here https://nextjs.org/learn/basics/create-nextjs-app?utm_source=next-site&utm_medium=homepage-cta&utm_campaign=next-website
Using windows 10 Pro with latest updates
Node v14.16.0
Yarn 1.22.10

Changes to index.js trigger recompilation, but browser does not update, no messages in the console, refresh in browser loads the page with the edits

**Comment by joao-pedrozo** [comment]
I'm wondering if maybe it's a low-level thing involving directly the hardware. I was having this issue using Ubuntu 20.04 then I changed to windows using WSL(still with Ubuntu) and sometimes the behaviour keeps happening. My processor is an AMD Ryzen 5 1600

**Comment by timotheegloerfeld** [comment]
Same issue here, also using wsl2 ubuntu and node 14. It works on 10.0.5 and stops working on 10.0.6, but I'm not sure which changes may have impacted this.

**Comment by ghost** [comment]
I have the same issue and I am so confused. No idea what's causing this at all. I tried to look at the changes between v10.0.5 and v10.0.6, but I can't figure it out. Looks like webpack 5 support was added in v10.0.6, but not sure if it's related to this issue in any way.

**Comment by ghost** [comment]
Hi all, happy to report that I have a solution!

## Solution

You need to start `next dev` on host `::` (the default unspecified IP address for IPv6):

```
next dev -H ::
```

Tested with Next.js v10.2.3 via WSL and Powershell using Chrome. ✅ Firefox doesn't work (sorry ☹).

## Explanation

Next.js defaults to the default unspecified IP address for IPv4 (`0.0.0.0`), but that does not seem compatible with Windows 10.

https://github.com/vercel/next.js/blob/28501c6af4270937f67c460c1358ac4d3e489b43/packages/next/cli/next-start.ts#L53

As others have pointed out, the bug was introduced as of Next v10.0.6. A quick `git blame` on `/packages/next/cli/next-start.ts` shows that the default IP address of `0.0.0.0` was set in https://github.com/vercel/next.js/pull/20409, which was indeed part of the PRs merged for v10.0.6:

![image](https://user-images.githubusercontent.com/55237550/119893145-51209e80-bf3b-11eb-945f-1972972ca7b8.png)

Mystery solved! 🎉🎉🎉

**Reply by bfunc** [reply]
Thanks a lot, it solves the issue

**Reply by thomasnguyen95** [reply]
Still a useful fix

**Comment by 3e33** [comment]
Try creating a file called ".env.local" in your root project folder and add "WATCHPACK_POLLING=true" to it. Restart the server.

**Comment by ttsirkia** [comment]
This seems to be still an issue even with version 11.0.1. Just spent some time to debug the problem until I found this discussion. The trick mentioned by @finnwoelm-sdsn is still valid. Works perfectly with that, but just with Chrome and using `localhost:3000`, not `127.0.0.1:3000`.

Is there any open issue related to this? Would be very helpful that the workaround and limitation would be mentioned in the documentation if this is the only way to get the feature working in Windows.

**Comment by ttsirkia** [comment]
It seems that this will be fixed in 11.0.2, https://github.com/vercel/next.js/pull/27306

**Comment by ttsirkia** [comment]
The mentioned fix was part of version 11.1.0. Now it works out of the box with Chrome but Firefox has still the same problem, for some reason.

**Comment by BirajAd** [comment]
looks like next doesn't watch file names that don't start with uppercase. My ```Index.js ``` file was named as ```index.js``` but as I changed it to ```Index.js``` with uppercase I, it worked.

**Comment by Civil-t** [comment]
Having the same issue on Windows 11
